### ISSUE TITLE
public variables are not expanded correctly when contract is imported

### ISSUE THREAD

**Issue by AntonD3** [vyperlang/vyper#3052]
### Version Information

* vyper Version (output of `vyper --version`): 0.3.3

### What's your issue about?

```
KeyError: 'name'
[14547] Failed to execute script 'vyper_compile' due to unhandled exception!

```

#### Sources

Foo.vy:
```
name: public(String[10])

@external
def __init__():
    self.name = "Foo"
```

Import.vy:
```
import Foo as Foo

# Test Foo.sol by getting it's name.
@external
@view
def getFooName(foo: address) -> String[10]:
    return Foo(foo).name()

```

**Comment by cairoeth** [comment]
I believe it was fixed in #2998 (for v0.3.4). @charles-cooper

**Comment by charles-cooper** [comment]
@cairoeth confirmed fixed as of e7733e15003816d7a323476678a2ac7b3755628f. (I have not verified the specific commit it was fixed in)
